pallets/public-credentials/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+
+		/// Removes the information pertaining a public credential from the
+		/// chain.
+		///
+		/// The preconditions of the `attestation::remove()` function must all
+		/// be fulfilled, as this function calls `attestation::remove()`
+		/// internally. Nevertheless, the opposite is not true.
+		/// Removing an attestation from the attestation pallet still requires
+		/// the attester to also remove any traces of the corresponding public
+		/// credential from this pallet.
+		///
+		/// The removal of the credential does not delete it entirely from the
+		/// blockchain history, but only its link *from* the blockchain state
+		/// *to* the blockchain history is removed.
+		///
+		/// Clients parsing public credentials should consider should interpret
+		/// the lack of such a link as the fact that the credential has been
+		/// removed by its attester some time in the past.
+		///
+		/// This function fails if a credential already exists for the specified
+		/// subject, regardless of the identity of the attester.
+		///
+		/// Emits `CredentialRemoved`.
+		#[pallet::weight({
+			let ac_weight = authorization.as_ref().map(|ac| ac.can_attest_weight()).unwrap_or(0);
+			<T as pallet::Config>::WeightInfo::remove().saturating_add(ac_weight)
+		})]
+		pub fn remove(
+			origin: OriginFor<T>,
+			claim_hash: ClaimHashOf<T>,
+			authorization: Option<T::AccessControl>,
+		) -> DispatchResultWithPostInfo {
+			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
+			let attester = source.subject();
+
+			// Verify that the credential exists
+			let credential_subject =
+				CredentialsUnicityIndex::<T>::get(&claim_hash).ok_or(Error::<T>::CredentialNotFound)?;
+			// Should never happen if the line above succeeds
+			let credential_entry =
+				Credentials::<T>::get(&credential_subject, &claim_hash).ok_or(Error::<T>::InternalError)?;
+
+			// Delegate to the attestation pallet the removal logic
+			// This guarantees that the authorized attester is calling this function
+			let result = attestation::Pallet::<T>::remove_attestation(attester, claim_hash, authorization)?;
+
+			Self::remove_credential_entry(credential_subject, claim_hash, credential_entry);
+
+			Ok(result)
+		}
+
+		/// Performs the same function as the `remove` extrinsic, with the
+		/// difference that the caller of this function must be the original
+		/// payer of the deposit, and not the original attester of the
+		/// credential.
+		///
+		/// It calls `attestation::reclaim_deposit()` internally, nevertheless
+		/// the opposite is not true. Removing an attestation from the
+		/// attestation pallet still requires the deposit payer to also remove
+		/// any traces of the corresponding public credential from this pallet.
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::reclaim_deposit())]
+		pub fn reclaim_deposit(origin: OriginFor<T>, claim_hash: ClaimHashOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// Verify that the credential exists
```

Comment:
C: Maybe move to separate function.

Reply:
https://github.com/KILTprotocol/kilt-node/pull/378/commits/8483968447060c3ef26bf871df82571c01e52e2e